.github/workflows/galata.yml

```diff
@@ -84,3 +85,121 @@ jobs:
         if: always()
         run: |
           cat /tmp/jupyterlab_server.log
+
+  test-documentation:
+    name: Visual Regression Documentation
+    # Python version is frozen through strategy.matrix.python-version
+    # Python dependencies are frozen in the installation step
+    # Note: IPython is among the frozen package
+    timeout-minutes: 80
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        # Freeze Python version because it appears in console header
+        python-version: ['3.9.9']
+    steps:
+      - name: Checkout JupyterLab
+        uses: actions/checkout@v2
+        with:
+          path: core
+
+      - name: Checkout demo project
+        uses: actions/checkout@v2
+        with:
+          repository: jupyterlab/jupyterlab-demo
+          ref: master
+          path: demo
+
+      - name: Get demo folder
+        run: |
+          echo "DEMO_DIR=$PWD" >> $GITHUB_ENV
+        working-directory: demo
+
+      - name: Base Setup
+        uses: jupyterlab/maintainer-tools/.github/actions/base-setup@v1
+
+      - name: Install dependencies
+        run: |
+          set -ex
+          # Install chinese font
+          sudo apt-get update
+          sudo apt-get install fonts-dejavu fonts-noto
+
+          bash ./scripts/ci_install.sh
+          # Freeze the packages to ensure consistent look and feel
+          # IPython is frozen because its version is displayed in
+          # the console header
+          pip install altair==4.2.0 ipython==8.0.0 ipywidgets==7.6.5 jupyterlab-geojson==3.2.0 jupyterlab-language-pack-zh-CN==3.2.post7 matplotlib==3.5.1 pandas==1.3.5 scipy==1.7.3 vdom==0.6 vega_datasets==0.9.0
```

Comment:
Wondering whether we should extract these frozen dependencies somewhere else in a separate file, so it's easier to keep track of them?

Reply:
What about adding an additional extra in setup.cfg? We could probably drop `docs/environment.yml` and add two extras like: `doc` and `doc-screenshots`.

Reply:
Sounds like extras would work.

Don't we need to keep `docs/environment.yml` for ReadTheDocs? 

https://github.com/jupyterlab/jupyterlab/blob/1903d0dd09b2415a998dd46de15b54a5c8988db5/readthedocs.yml#L4-L5

Reply:
We could switch to [pip](https://docs.readthedocs.io/en/stable/config-file/v2.html#python) and specify that nodejs is a need build tool.